lms/djangoapps/discussion_api/tests/test_api.py

```diff
@@ -1579,3 +1583,124 @@ def test_invalid_field(self):
             assertion.exception.message_dict,
             {"raw_body": ["This field is required."]}
         )
+
+
+@ddt.ddt
+class DeleteThreadTest(CommentsServiceMockMixin, UrlResetMixin, ModuleStoreTestCase):
+    """Tests for delete_thread"""
+    @mock.patch.dict("django.conf.settings.FEATURES", {"ENABLE_DISCUSSION_SERVICE": True})
+    def setUp(self):
+        super(DeleteThreadTest, self).setUp()
+        httpretty.reset()
+        httpretty.enable()
+        self.addCleanup(httpretty.disable)
+        self.user = UserFactory.create()
+        self.register_get_user_response(self.user)
+        self.request = RequestFactory().get("/test_path")
+        self.request.user = self.user
+        self.course = CourseFactory.create()
+        self.thread_id = "test_thread"
+        CourseEnrollmentFactory.create(user=self.user, course_id=self.course.id)
+
+    def register_thread(self, overrides=None):
+        """
+        Make a thread with appropriate data overridden by the overrides
+        parameter and register mock responses for both GET and DELETE on its
+        endpoint.
+        """
+        cs_data = make_minimal_cs_thread({
+            "id": self.thread_id,
+            "course_id": unicode(self.course.id),
+            "user_id": str(self.user.id),
+        })
+        cs_data.update(overrides or {})
+        self.register_get_thread_response(cs_data)
+        self.register_delete_thread_response(cs_data["id"])
+
+    def test_basic(self):
+        self.register_thread()
+        self.assertIsNone(delete_thread(self.request, self.thread_id))
+        self.assertEqual(
+            urlparse(httpretty.last_request().path).path,
+            "/api/v1/threads/{}".format(self.thread_id)
+        )
+        self.assertEqual(httpretty.last_request().method, "DELETE")
+
+    def test_thread_id_not_found(self):
+        self.register_get_thread_error_response("missing_thread", 404)
+        with self.assertRaises(Http404):
+            delete_thread(self.request, "missing_thread")
+
+    def test_nonexistent_course(self):
+        self.register_thread({"course_id": "not_a_real_course"})
+        with self.assertRaises(InvalidKeyError):
+            delete_thread(self.request, self.thread_id)
```

Comment:
Hmm, I should rename the test then to something like test_course_with_no_access. This edge case would be if a user had created a thread and then became unenrolled.